frontend/src/metabase/visualizations/visualizations/SmartScalar/compute.js

```diff
@@ -1,206 +1,475 @@
-  MISSING: "PREVIOUS_VALUE_MISSING",
-  SAME: "PREVIOUS_VALUE_SAME",
-  CHANGED: "PREVIOUS_VALUE_CHANGED",
-};
+function computeTrendPeriodsAgo({ currentMetricData, series, settings }) {
+  const [
+    {
+      data: { rows },
+    },
+  ] = series;
 
-function computeChangeTypeWithOptions({
-  formatOptions,
-  percentChange,
-  prevValue,
-}) {
-  if (isEmpty(prevValue)) {
-    return {
-      comparisonType: PREVIOUS_VALUE_OPTIONS.MISSING,
-      percentChangeStr: t`N/A`,
-      prevValueStr: t`(empty)`,
-    };
+  const {
+    date,
+    dateUnitSettings,
+    indexData: { dimensionColIndex, metricColIndex, latestRowIndex },
+  } = currentMetricData;
+
+  if (isEmpty(dateUnitSettings.dateUnit)) {
+    throw Error("No date unit supplied for periods ago comparison");
   }
 
-  if (percentChange === 0) {
+  const { type, value } = settings["scalar.comparisons"];
+  if (type === COMPARISON_TYPES.PERIODS_AGO && !Number.isInteger(value)) {
+    throw Error("No integer value supplied for periods ago comparison");
+  }
+  const dateUnitsAgo = value ?? 1;
+
+  return computeComparisonPeriodsAgo({
+    rows,
+    dimensionColIndex,
+    metricColIndex,
+    nextValueRowIndex: latestRowIndex,
+    nextDate: date,
+    dateUnitSettings,
+    dateUnitsAgo,
+  });
+}
+
+function computeComparisonPeriodsAgo({
+  rows,
+  dimensionColIndex,
+  metricColIndex,
+  nextValueRowIndex,
+  nextDate,
+  dateUnitSettings,
+  dateUnitsAgo,
+}) {
+  const dateUnitDisplay = Lib.describeTemporalUnit(
+    dateUnitSettings.dateUnit,
+  ).toLowerCase();
+
+  const prevDate = moment
+    .parseZone(nextDate)
+    .subtract(dateUnitsAgo, dateUnitSettings.dateUnit)
+    .format();
+
+  const comparisonDescStr =
+    dateUnitsAgo === 1
+      ? t`vs. previous ${dateUnitDisplay}`
+      : computeComparisonStrPreviousValue({
+          dateUnitSettings,
+          nextDate,
+          prevDate,
+        });
+
+  const rowPeriodsAgo = getRowOfPeriodsAgo({
+    prevDate,
+    dateUnit: dateUnitSettings.dateUnit,
+    dateUnitsAgo,
+    dimensionColIndex,
+    nextValueRowIndex,
+    rows,
+  });
+  // if no row exists with date "X periods ago"
+  if (isEmpty(rowPeriodsAgo)) {
     return {
-      comparisonType: PREVIOUS_VALUE_OPTIONS.SAME,
-      percentChangeStr: t`No change`,
-      prevValueStr: "",
+      comparisonDescStr,
     };
   }
 
+  const prevValue = rowPeriodsAgo[metricColIndex];
+
   return {
-    comparisonType: PREVIOUS_VALUE_OPTIONS.CHANGED,
-    changeArrowIconName: percentChange < 0 ? "arrow_down" : "arrow_up",
-    percentChangeStr: formatChange(percentChange),
-    prevValueStr: formatValue(prevValue, formatOptions),
+    comparisonDescStr,
+    comparisonValue: prevValue,
   };
 }
 
-function computePreviousPeriodComparison({
+function getRowOfPeriodsAgo({
+  prevDate,
+  dateUnit,
+  dateUnitsAgo,
+  dimensionColIndex,
+  nextValueRowIndex,
   rows,
-  metricIndex,
-  dimensionIndex,
-  nextValue,
+}) {
+  const targetDate = moment.parseZone(prevDate);
```

Comment:
We are moving to `dayjs` so it would be nice to use it here instead of moment. Probably, should be easy to replace

Reply:
I attempted this, but it seems that in the use case I have, moment and dayjs have significantly different behaviors. Dayjs has weird behaviors that do not seem to have been fixed yet (https://github.com/iamkun/dayjs/issues/651, https://github.com/iamkun/dayjs/pull/2060).

**Not Preserving UTC Offsets**

![image](https://github.com/metabase/metabase/assets/22608765/7c2535aa-4141-462c-b03e-1204d8025b63)
![image](https://github.com/metabase/metabase/assets/22608765/0e8927a7-8556-493f-a85e-f03d6bc57030)

**Changing UTC Offsets Based on User Time-zone**

_Timezone: UTC -05:00_
![image](https://github.com/metabase/metabase/assets/22608765/21287b1b-2258-450c-b30c-e915fd81a9f1)
![image](https://github.com/metabase/metabase/assets/22608765/0f0125ad-7bb2-43f2-aeda-e16220b4cf78)

_Timezone: UTC +08:00_
![image](https://github.com/metabase/metabase/assets/22608765/022b6475-b658-43ce-aaee-34bc433f9019)
![image](https://github.com/metabase/metabase/assets/22608765/77c2db27-ad7a-467a-ad03-844f9a2c38be)

---

@alxnddr WDYT?